addons/web_editor/static/lib/odoo-editor/src/commands/commands.js

```diff
@@ -505,15 +506,31 @@ export const editorCommands = {
         const selectedNodes = getSelectedNodes(editor.editable);
         const fonts = selectedNodes.flatMap(node => {
             let font = closestElement(node, 'font');
-            const children = font && [...font.childNodes];
+            const children = font && getDescendants(font);
             if (font && font.nodeName === 'FONT') {
                 // Partially selected <font>: split it.
                 const selectedChildren = children.filter(child => selectedNodes.includes(child));
                 if (selectedChildren.length) {
-                    const after = selectedChildren[selectedChildren.length - 1].nextSibling;
-                    font = after ? splitElement(font, childNodeIndex(after))[0] : font;
-                    const before = selectedChildren[0].previousSibling;
-                    font = before ? splitElement(font, childNodeIndex(before) + 1)[1] : font;
+                    let after = selectedChildren[selectedChildren.length - 1].nextSibling;
+                    let newFont = font;
+                    // Split up ancestors up to font
+                    while (after && after.parentElement !== font) {
+                        newFont = splitElement(after.parentElement, childNodeIndex(after))[0];
+                        after = newFont.nextSibling;
+                    }
+                    if (after) {
+                        font = splitElement(font, childNodeIndex(after))[0];
+                    }
+                    let before = selectedChildren[0].previousSibling;
+                    while (before && before.parentElement !== font) {
+                        newFont = splitElement(before.parentElement, childNodeIndex(before) + 1)[1];
+                        before = newFont.previousSibling;
+                    }
+                    if (before) {
+                        font = splitElement(font, childNodeIndex(before) + 1)[1];
+                    }
+                } else {
+                    font = [];
```

Comment:
I would create a generic function in util rather than specifically doing it for this case as we might need it for other purpose.

Reply:
Why not but <hat would the generic function be for exactly then?

Reply:
To split all selected nodes up to a particular node and would be called `splitSelectedNode`.
If you think we will not have other use case, then no need to create the function.

Reply:
Yeah that makes sense I'll do that thanks